Adding Glances & qDirstat to Tipi!

itsrllyhim opened the pull request:
Please see 'Read Me'.

DrMxrcy commented on apps/glances/config.json:
```diff
@@ -0,0 +1,19 @@
+  "name": "Glances",
+  "port": 8420,
+  "available": true,
+  "exposable": true,
```
```suggestion
  "exposable": false,
```
There should be no reason to expose this with a Public Domain IMO. Can confirm with @meienberger

itsrllyhim replied:
Sounds good, no disagreement.

DrMxrcy commented on apps/glances/docker-compose.yml:
```diff
@@ -0,0 +1,42 @@
+            traefik.enable: true
+            traefik.http.middlewares.glances-web-redirect.redirectscheme.scheme: https
+            traefik.http.services.glances.loadbalancer.server.port: 61208
+            # Web
+            traefik.http.routers.glances-insecure.rule: Host(`${APP_DOMAIN}`)
+            traefik.http.routers.glances-insecure.entrypoints: web
+            traefik.http.routers.glances-insecure.service: glances
+            traefik.http.routers.glances-insecure.middlewares: glances-web-redirect
+            # Websecure
+            traefik.http.routers.glances.rule: Host(`${APP_DOMAIN}`)
+            traefik.http.routers.glances.entrypoints: websecure
+            traefik.http.routers.glances.service: glances
+            traefik.http.routers.glances.tls.certresolver: myresolver
+            # Local domain
+            traefik.http.routers.glances-local-insecure.rule: Host(`glances.${LOCAL_DOMAIN}`)
+            traefik.http.routers.glances-local-insecure.entrypoints: web
+            traefik.http.routers.glances-local-insecure.service: glances
+            traefik.http.routers.glances-local-insecure.middlewares: glances-web-redirect
+            # Local domain secure
+            traefik.http.routers.glances-local.rule: Host(`glances.${LOCAL_DOMAIN}`)
+            traefik.http.routers.glances-local.entrypoints: websecure
+            traefik.http.routers.glances-local.service: glances
+            traefik.http.routers.glances-local.tls: true
```
```suggestion
            # Local domain
            traefik.http.routers.glances-local-insecure.rule: Host(`glances.${LOCAL_DOMAIN}`)
            traefik.http.routers.glances-local-insecure.entrypoints: web
            traefik.http.routers.glances-local-insecure.service: glances
            traefik.http.routers.glances-local-insecure.middlewares: glances-web-redirect
            # Local domain secure
            traefik.http.routers.glances-local.rule: Host(`glances.${LOCAL_DOMAIN}`)
            traefik.http.routers.glances-local.entrypoints: websecure
            traefik.http.routers.glances-local.service: glances
            traefik.http.routers.glances-local.tls: true
```

In accordance with my Suggestion Above

itsrllyhim replied:
Cheers, thanks!

DrMxrcy reviewed (commented):
Looks Good. Deployed well, Just the removal of the exposable domain, without any Authentication in front of it.

itsrllyhim commented:
Alrighty that was my first attempt at doing this by my lonesome so it took a couple of tries to pass the tests- if someone (looking at you here @DrMxrcy haha) could double check my work, I would greatly appreciate it!

itsrllyhim commented:
Okay! Another update on qDirStat: Initially passed the tests but failed to install to tipi. I corrected the issue causing that (Config directory had to match a specific constraint), I also fixed a minor issue with the ReadMe not loading the correct image for a screenshot. 

**Problems:**

I can't seem to get it to scan outside of the config directory. The documentations provided an ENV variable that didn't seem to have an impact even when overridden (since removed from the docker-compose.yaml). It also seems the entire tool runs inside of a Kasm machine, where if the user closes out of the program, they then become sort of "stuck" and have to manually stop and restart the app to get it back up.  I am in touch with the LinuxServer.io folks to see if we can find a solution- if so, will update the image.

meienberger commented on apps/glances/docker-compose.yml:
```diff
@@ -0,0 +1,32 @@
+        restart: unless-stopped
+        ports:
+            - ${APP_PORT}:61208
+            - 8421:61209
```
For what is this port needed? I removed it and app launched the same

itsrllyhim replied:
Honestly, not particularly sure- Steve and I just followed the docs. If it works without it, then there's no need for it. Feel free to remove it!

meienberger commented on apps/qdirstat/docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+            - ${APP_PORT}:3000
+        volumes:
+            - ${APP_DATA_DIR}/data/qdirstat/config
+            - ${APP_DATA_DIR}/data/qdirstat/app/data
```
I think this app would be more useful if we could see the whole system tree from `/` what do you think?

itsrllyhim replied:
You're right, I think that would solve the problem I alluded to [here](https://github.com/meienberger/runtipi-appstore/pull/1161#issuecomment-1710990364). Feel free to make the change, or ping me if you would like me to!
